src/types/BaseFieldLocalization.ts

```diff
@@ -0,0 +1,103 @@
+import { ajv } from '../ajv';
+import type { JSONSchemaType } from 'ajv';
+import type { Writable } from './Writable';
+
+interface BaseFieldLocalization {
+	baseFieldId: number;
```

Comment:
The solution to your type woes is to make this and language `readonly` -- if they are the primary key then they are no longer writable attributes.

I think with this you can address the issues both here AND in the openAPI spec, and remove the need for a `WithBaseFieldAndLocalizationContext` type.